mods/cnc/maps/gdi41/gdi41.lua

```diff
@@ -0,0 +1,273 @@
+
+	InitObjectives(Player1)
+	ClearPath = AddPrimaryObjective(Player1, "clear-path")
+	EliminateAllNod = AddPrimaryObjective(Player1, "eliminate-nod")
+
+	InitAI()
+	CheckObjectives()
+
+	Flare = Actor.Create("flare", true, { Owner = Player1, Location = DefaultFlareLocation.Location })
+
+	Trigger.AfterDelay(1,function()
+		AmbushTeam = Map.ActorsInBox(Map.CenterOfCell(CPos.New(46,5)), Map.CenterOfCell(CPos.New(60,53)), function(a) return a.Owner == EnemyAi and a.Type ~= "stnk" end)
+		Trigger.OnAllKilled(AmbushTeam, function()
+			Player1.MarkCompletedObjective(ClearPath)
+			Trigger.AfterDelay(50, function()
+				Trigger.AfterDelay(750, function()
+					Flare.Destroy()
+				end)
+				Media.PlaySpeechNotification(Player1, "Reinforce")
+				Reinforcements.Reinforce(Player1, MCVReinf, { CPos.New(56,2), waypoint0.Location }, 25, function(a)
+					a.Move(waypoint1.Location, 2)
+					a.Move(waypoint2.Location, 2)
+					a.Move(CPos.New(49,44), 2)
+					if a.HasProperty("Deploy") then
+						a.Move(CPos.New(48,51))
+						a.Deploy()
+					end
+				end)
+				Utils.Do(ClearPathCameras, function(c)
+					c.Destroy()
+				end)
+			end)
+		end)
+	end)
+
+	NodAirSupport = Actor.Create("Astk.proxy", true, { Owner = EnemyAi })
+	Trigger.AfterDelay(DateTime.Seconds(AstkDelay), function()
+		SendNodAirstrike(true)
+		Trigger.AfterDelay(DateTime.Seconds(20), function()
+			Media.DisplayMessage(UserInterface.Translate("air-strikes-intel-report"))
+			Trigger.AfterDelay(DateTime.Seconds(5), function()
+				CaptureCommCenter = AddSecondaryObjective(Player1, "capture-nod-communications-center")
+			end)
+		end)
+	end)
+
+	Trigger.OnKilledOrCaptured(NodOutPostCYard, function()
+		SendNodAirstrike(false)
+	end)
+
+	Trigger.OnAllKilled(NodSams, function()
+		AirSupport = Actor.Create("Astk.proxy", true, { Owner = Player1 })
+		Player1.MarkCompletedObjective(DestroySams)
+	end)
+
+	Utils.Do(NodGuns, function(g)
+		Trigger.OnKilled(g, function()
+			SendEngTeam()
+			Utils.Do(NodGuns, function(gun) if not gun.IsDead then Trigger.Clear(gun, "OnKilled") end end)
+		end)
+	end)
+
+	Trigger.OnEnteredFootprint(CenterNodBaseEntrance, function(a, id)
+		if a.Owner == Player1 then
+			local cargo = Reinforcements.ReinforceWithTransport(EnemyAi, "tran", { "rmbo" }, { CPos.New(0,32), waypoint12.Location }, { CPos.New(0,32) })[2]
+			Trigger.OnIdle(cargo[1], function()
+				Trigger.Clear(cargo[1], "OnIdle")
+				Media.PlaySpeechNotification(Player1, "BaseAttack")
+				Trigger.AfterDelay(5, function() GoDemolitionMan(cargo[1]) end)
+			end)
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.OnDamaged(NodCYard, function(_, atk, _)
+		if atk.Owner == Player1 and not NukeLaunched then
+			NukeLaunched = true
+			LaunchNuke(false)
+			Trigger.Clear(NodCYard, "OnDamaged")
+			Trigger.AfterDelay(1, function()
+				RepairBuilding(EnemyAi, NodCYard, 0.75)
+				NodCYard.StartBuildingRepairs()
+			end)
+		end
+	end)
+
+	Trigger.OnKilled(NodTmpl, function()
+		NukeDelay = 0
+	end)
```

Comment:
It's possible to capture the temple before the nuke countdown (or yard trap) kicks in. If so, the player can get nuked by their newly captured temple. I'll admit this was funny to see happen.
```suggestion
	Trigger.OnKilledOrCaptured(NodTmpl, function()
		NukeDelay = 0
	end)
```
An owner check in `LaunchNuke` should accomplish the same thing if that's easier to read.